services/queue/src/queueservice.js

```diff
@@ -47,204 +36,54 @@ const PRIORITIES = [
-    this.resolvedQueueReady = null;
-
-    // Store deadlineQueue name, and remember if we've created it
-    this.deadlineQueue = options.deadlineQueue;
+    this.db = options.db;
     this.deadlineDelay = options.deadlineDelay;
-    this.deadlineQueueReady = null;
-
-    // Keep a cache of pending counts as mapping:
-    //    <provisionerId>/<workerType> -> {lastUpdated, count: promise}
-    this.countPendingCache = {};
   }
 
   terminate() {
-    clearInterval(this.queueResetInterval);
-  }
-
-  _putMessage(queue, message, { visibility, ttl, taskQueueId, priority }) {
-    let text = Buffer.from(JSON.stringify(message)).toString('base64');
-    return this.monitor.timer('putMessage', this.client.putMessage(queue, text, {
-      visibilityTimeout: visibility,
-      messageTTL: ttl,
-      taskQueueId,
-      priority,
-    }));
-  }
-
-  async _getMessages(queue, { visibility, count }) {
-    let messages = await this.monitor.timer('getMessages', this.client.getMessages(queue, {
-      visibilityTimeout: visibility,
-      numberOfMessages: count,
-    }));
-    return messages.map(msg => {
-      return {
-        payload: JSON.parse(Buffer.from(msg.messageText, 'base64')),
-        remove: this.client.deleteMessage.bind(
-          this.client,
-          queue,
-          msg.messageId,
-          msg.popReceipt,
-        ),
-        release: this.client.updateMessage.bind(
-          this.client,
-          queue,
-          msg.messageText,
-          msg.messageId,
-          msg.popReceipt, {
-            visibilityTimeout: 0,
-          },
-        ),
-      };
-    });
-  }
-
-  /** Ensure existence of the claim queue */
-  ensureClaimQueue() {
-    if (this.claimQueueReady) {
-      return this.claimQueueReady;
-    }
-    let ready = this.client.createQueue(this.claimQueue).catch(err => {
-      // Don't cache negative results
-      this.claimQueueReady = null;
-      throw err;
-    });
-    return this.claimQueueReady = ready;
-  }
-
-  /** Ensure existence of the resolved task queue */
-  ensureResolvedQueue() {
-    if (this.resolvedQueueReady) {
-      return this.resolvedQueueReady;
-    }
-    let ready = this.client.createQueue(this.resolvedQueue).catch(err => {
-      // Don't cache negative results
-      this.resolvedQueueReady = null;
-      throw err;
-    });
-    return this.resolvedQueueReady = ready;
-  }
-
-  /** Ensure existence of the deadline queue */
-  ensureDeadlineQueue() {
-    if (this.deadlineQueueReady) {
-      return this.deadlineQueueReady;
-    }
-    let ready = this.client.createQueue(this.deadlineQueue).catch(err => {
-      // Don't cache negative results
-      this.deadlineQueueReady = null;
-      throw err;
-    });
-    return this.deadlineQueueReady = ready;
   }
 
   /** Enqueue message to become visible when claim has expired */
-  async putClaimMessage(taskId, runId, takenUntil) {
+  async putClaimMessage(taskId, runId, takenUntil, taskQueueId, workerGroup, workerId) {
```

Comment:
this class is only being adjusted to new functions after azqueue library was removed